Content.Shared/Actions/ActionContainerSystem.cs

```diff
@@ -328,50 +324,7 @@ private void OnEntityRemoved(EntityUid uid, ActionsContainerComponent component,
         if (!_actions.TryGetActionData(args.Entity, out var data, false))
             return;
 
-        // No event - the only entity that should care about this is the entity that the action was provided to.
-        if (data.AttachedEntity != null)
-            _actions.RemoveAction(data.AttachedEntity.Value, args.Entity, null, data);
-
-        var ev = new ActionRemovedEvent(args.Entity, data);
-        RaiseLocalEvent(uid, ref ev);
         data.Container = null;
-    }
-
-    private void OnActionAdded(EntityUid uid, ActionsContainerComponent component, ActionAddedEvent args)
-    {
-        if (TryComp<MindComponent>(uid, out var mindComp) && mindComp.OwnedEntity != null && HasComp<ActionsContainerComponent>(mindComp.OwnedEntity.Value))
-            _actions.GrantContainedAction(mindComp.OwnedEntity.Value, uid, args.Action);
-    }
-}
-
-/// <summary>
-/// Raised directed at an action container when a new action entity gets inserted.
-/// </summary>
-[ByRefEvent]
-public readonly struct ActionAddedEvent
-{
-    public readonly EntityUid Action;
-    public readonly BaseActionComponent Component;
-
-    public ActionAddedEvent(EntityUid action, BaseActionComponent component)
-    {
-        Action = action;
-        Component = component;
-    }
-}
-
-/// <summary>
-/// Raised directed at an action container when an action entity gets removed.
-/// </summary>
-[ByRefEvent]
-public readonly struct ActionRemovedEvent
-{
-    public readonly EntityUid Action;
-    public readonly BaseActionComponent Component;
-
-    public ActionRemovedEvent(EntityUid action, BaseActionComponent component)
-    {
-        Action = action;
-        Component = component;
```

Comment:
I removed ActionAddedEvent / ActionRemovedEvent because they're largely unused and rely upon the container side of actions when I think this should be done on the action side.

Normally I'm in support of having subscribing to container changes and handling things that way (e.g. body code) because it's much more robust against prediction or other issues, but in this case I'm not familiar enough with actions code to make that call.

Reply:
This is needed for spells and such the last time I tested.

Reply:
I moved the grant action so on master it should more or less be parity if they're using addaction / removeaction, though if there were some way to playtest containers it would be better.

Reply:
I could try pulling this into my branch to see

Reply:
Working on magic refactor again and I will need the ActionAdded and ActionRemoved events.